fix(transformer/typescript): redeclaration of `namespace` with `enum` causes syntax error

Dunqing opened the pull request:
Fix https://github.com/rolldown/rolldown/pull/4003/files/0235d47b4f7a589fd1d6fd38df776b261a9f5875#r2023968962


Input:
```ts
namespace x { console.log(x, y) }
enum x { y = 123 }
``` 

Output diff compare before:
```diff
let x;
(function(_x) {
  console.log(x, y);
})(x || (x = {}));
-var x = /* @__PURE__ */ function(x) {
+x = /* @__PURE__ */ function(x) {
  x[x["y"] = 123] = "y";
  return x;
}(x || {});
```

`namespace` transformation has already inserted `let x`, so in the enum transformation we don't need to create an `var x` binding, instead we just refer to `x`.

BTW In this case, `Babel` has a syntax error "Duplicate declaration "x"", see [Babel Playground](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=HYQwtgpgzgDiDGEAEAPJBvJ8D2wrYBsIA6A7AcwAoUAaJATwEokBfAKAmAFcxUMGkAXiQBGAEwBmVkA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=typescript&prettier=true&targets=&version=7.27.0&externalPlugins=%40babel%2Fplugin-external-helpers%407.25.9%2C%40babel%2Fplugin-transform-class-properties%407.25.9%2C%40babel%2Fplugin-syntax-jsx%407.25.9&assumptions=%7B%7D), but `TypeScript` can compile it

Dunqing commented:
* **#10172** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10172?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10172?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10172).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Apr 2, 2:39 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(transformer/typescript): redeclaration of `namespace` with `enum` causes syntax error (#10172)

Fix https://github.com/rolldown/rolldown/pull/4003/files/0235d47b4f7a589fd1d6fd38df776b261a9f5875#r2023968962

Input:
```ts
namespace x { console.log(x, y) }
enum x { y = 123 }
```

Output diff compare before:
```diff
let x;
(function(_x) {
  console.log(x, y);
})(x || (x = {}));
-var x = /* @__PURE__ */ function(x) {
+x = /* @__PURE__ */ function(x) {
  x[x["y"] = 123] = "y";
  return x;
}(x || {});
```

`namespace` transformation has already inserted `let x`, so in the enum transformation we don't need to create an `var x` binding, instead we just refer to `x`.

BTW In this case, `Babel` has a syntax error "Duplicate declaration "x"", see [Babel Playground](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=HYQwtgpgzgDiDGEAEAPJBvJ8D2wrYBsIA6A7AcwAoUAaJATwEokBfAKAmAFcxUMGkAXiQBGAEwBmVkA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=typescript&prettier=true&targets=&version=7.27.0&externalPlugins=%40babel%2Fplugin-external-helpers%407.25.9%2C%40babel%2Fplugin-transform-class-properties%407.25.9%2C%40babel%2Fplugin-syntax-jsx%407.25.9&assumptions=%7B%7D), but `TypeScript` can compile it
